ecal/core/src/ecal_process.cpp

```diff
@@ -366,7 +366,25 @@ namespace eCAL
 
     void SleepMS(const long time_ms_)
     {
-      std::this_thread::sleep_for(std::chrono::milliseconds(time_ms_));
+      #ifdef ECAL_OS_WINDOWS
+      {
+        Sleep(time_ms_);
+      }
+      #else
+        std::this_thread::sleep_for(std::chrono::milliseconds(time_ms_));
+      #endif
+    }
+
+    void SleepNS(const long long time_ns_)
+    {
+      #ifdef ECAL_OS_WINDOWS
+      {
+        auto milliseconds = time_ns_ / 1000000 + ((time_ns_ / 1000000) != 0);
```

Comment:
I assume the idea is to round up by checking if the division has a rest?

```suggestion
        auto milliseconds = time_ns_ / 1000000 + ((time_ns_ % 1000000) != 0);
```

Reply:
I am not 100% sure I agree with this function. We shouldn't offer a `SleepNS` function, if we cannot sleep to that precision.
And in the above case I think a `auto milliseconds = time_ns / 1000000` is fine, since it will automatically round down? or do you want to potentially round up? then I would look at `std` functions like `floor`, `ceil`, ....

Reply:
> I am not 100% sure I agree with this function. We shouldn't offer a `SleepNS` function, if we cannot sleep to that precision. 

It is common practice to provide those of functions in order to have a common API. They also exist in STL, even though many operating systems fail to reach that kind of precision. In this case, Win32 will not reach nanosecond precision, but Linux systems maybe will.

> And in the above case I think a `auto milliseconds = time_ns / 1000000` is fine, since it will automatically round down? or do you want to potentially round up?

Sleep functions are usually defined as sleeping "at least" for the given duration. Therefore, you need to round up, not down.

> then I would look at `std` functions like `floor`, `ceil`, ....

For those functions you need to convert to a floating point number, first.  I don't think that would be a good idea when you can just add a 1 as integer.

Reply:
@FlorianReimold you are right, it was a typo